### PR TITLE
Add basic skeleton of libpod runtime

### PR DESCRIPTION
**Pull request by mheon**
Still WIP, but required to get work on #803 moving.

### CODE REVIEW

**Comment by mheon** [comment]
Unit tests are coming, but I'm having trouble convincing the makefile to run them.

@rhatdan PTAL

**Comment by rhatdan** [comment]
@mrunalp @alexlarsson PTAL

File: libpod/options.go [hunk 1]
```diff
@@ -37,68 +35,146 @@ const (
...
 
 // WithSignaturePolicy specifies the path of a file which decides how trust is
 // managed for images we've pulled.
 // If this is not specified, the system default configuration will be used
 // instead
 func WithSignaturePolicy(path string) RuntimeOption {
-	return runtimeNotImplemented
+	return func(rt *Runtime) error {
+		if rt.valid {
+			return fmt.Errorf("runtime has already been finalized")
```
Comment (TomSweeneyRedHat) [inline libpod/options.go]: You've this same error verbiage in four or five functions.  Can you tweak them to at least be slightly different for each?

Comment (mheon) [reply libpod/options.go]: I think the error should probably remain the same (given it's the same error condition everywhere), but you make a good point that it's shared everywhere - I'll make it a variable to remove the duplication

File: libpod/runtime.go [hunk 2]
```diff
@@ -1,26 +1,135 @@
...
+
+	return config
+}
+
+// Shutdown shuts down the runtime and associated containers and storage
+// If force is true, containers and mounted storage will be shut down before
+// cleaning up; if force is false, an error will be returned if there are
+// still containers running or mounted
+func (r *Runtime) Shutdown(force bool) error {
+	r.lock.Lock()
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: Could we ever get into a deadlock here and not be able to shutdown?  I think this is OK as is, but thought I'd throw the question out there for consideration.  IDK if we might need a "super" shutdown that is triggered after waiting some configurable time or not.  I don't think that's needed, but don't know that for a fact.

Comment (mheon) [reply libpod/runtime.go]: I can definitely see the value of having a super-force flag that overrides the locking requirement and shuts everything down to ensure we can at least gracefully shutdown if we hang somewhere. I think we can hold off doing so for now, though - we don't actually have anything that can hang yet

**Comment by mrunalp** [comment]
This looks fine.

**Comment by rhatdan** [comment]
@mheon is this ready to merge?

**Comment by mheon** [comment]
@rhatdan Sure, I'll land the tests separately

**Comment by rhatdan** [comment]
I will merge this but could you take my patch set and show me how I am supposed to call it?

**Comment by mheon** [comment]
@rhatdan Sure, I can work up some examples of how to initialize Runtime and call into it.

**Pull request merged**